hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/HoodieStreamer.java

```diff
@@ -792,6 +793,10 @@ private void reInitDeltaSync() throws IOException {
     @Override
     protected Pair<CompletableFuture, ExecutorService> startService() {
       ExecutorService executor = Executors.newFixedThreadPool(1);
+      HoodieTableMetaClient meta = HoodieTableMetaClient.builder()
```

Comment:
MetaClient is expensive.. so having one just for `InstantFactory` feels like a stretch. we need to fix this (anything like this)